fem/fe/fe_pyramid.hpp

```diff
@@ -0,0 +1,489 @@
+
+   static DenseMatrix grad_mu01(real_t z);
+
+   static real_t mu0(real_t z, const Vector &xy, unsigned int ab)
+   { return 1.0 - xy[ab-1] / (1.0 - z); }
+   static real_t mu1(real_t z, const Vector &xy, unsigned int ab)
+   { return xy[ab-1] / (1.0 - z); }
+
+   static Vector grad_mu0(real_t z, const Vector xy, unsigned int ab);
+   static Vector grad_mu1(real_t z, const Vector xy, unsigned int ab);
+
+   static Vector mu01(real_t z, Vector xy, unsigned int ab)
+   { return Vector({mu0(z, xy, ab), mu1(z, xy, ab)}); }
+
+   static DenseMatrix grad_mu01(real_t z, Vector xy, unsigned int ab);
+   static Vector mu01_grad_mu01(real_t z, Vector xy, unsigned int ab);
+
+   static real_t nu0(real_t z, Vector xy, unsigned int ab)
+   { return 1.0 - xy[ab-1] - z; }
+   static real_t nu1(real_t z, Vector xy, unsigned int ab) { return xy[ab-1]; }
+   static real_t nu2(real_t z, Vector xy, unsigned int ab) { return z; }
+
+   static Vector grad_nu0(real_t z, const Vector xy, unsigned int ab);
+   static Vector grad_nu1(real_t z, const Vector xy, unsigned int ab);
+   static Vector grad_nu2(real_t z, const Vector xy, unsigned int ab);
+
+   static Vector nu01(real_t z, Vector xy, unsigned int ab)
+   { return Vector({nu0(z, xy, ab), nu1(z, xy, ab)}); }
+   static Vector nu12(real_t z, Vector xy, unsigned int ab)
+   { return Vector({nu1(z, xy, ab), nu2(z, xy, ab)}); }
+   static Vector nu012(real_t z, Vector xy, unsigned int ab)
+   { return Vector({nu0(z, xy, ab), nu1(z, xy, ab), nu2(z, xy, ab)}); }
+   static Vector nu120(real_t z, Vector xy, unsigned int ab)
+   { return Vector({nu1(z, xy, ab), nu2(z, xy, ab), nu0(z, xy, ab)}); }
+
+   static DenseMatrix grad_nu01(real_t z, Vector xy, unsigned int ab);
+   static DenseMatrix grad_nu012(real_t z, Vector xy, unsigned int ab);
+   static DenseMatrix grad_nu120(real_t z, Vector xy, unsigned int ab);
+
+   static Vector nu01_grad_nu01(real_t z, Vector xy, unsigned int ab);
+   static Vector nu12_grad_nu12(real_t z, Vector xy, unsigned int ab);
+   static Vector nu012_grad_nu012(real_t z, Vector xy, unsigned int ab);
+
+   /// Shifted and Scaled Legendre Polynomials
+   /** Implements a scaled and shifted set of Legendre polynomials
+
+         $P_i(x;t) = P_i(x / t) * t^i$
+
+       where @a t >= 0.0, @a x $\in [0,t]$, and $P_i$ is the shifted Legendre
+       polynomial defined on $[0,1]$ rather than the usual $[-1,1]$. The
+       entries stored in @a u correspond to the values of
+       $P_0$, $P_1$, ... $P_p$.
+
+       @a u must be at least @a p + 1 in length
+   */
+   static void CalcScaledLegendre(int p, real_t x, real_t t,
+                                  real_t *u);
+   static void CalcScaledLegendre(int p, real_t x, real_t t,
+                                  real_t *u, real_t *dudx, real_t *dudt);
+
+   static void CalcScaledLegendre(int p, real_t x, real_t t,
+                                  Vector &u);
+   static void CalcScaledLegendre(int p, real_t x, real_t t,
```

Comment:
It looks like this function is only used to compute the derivatives of the integrated Legendre functions.

Computing the derivatives of the integrated Legendre functions as described in https://github.com/mfem/mfem/pull/4460#discussion_r1976701476 is more computationally efficient and will allow this function to be removed